app/models/miq_server/worker_management/kubernetes.rb

```diff
@@ -215,9 +215,32 @@ def save_pod(pod)
 
     ch = Concurrent::Hash.new
     ch[:label_name]            = pod.metadata.labels.name
-    ch[:last_state_terminated] = pod.status.containerStatuses.any? { |cs| cs.lastState.terminated }
     ch[:container_restarts]    = pod.status.containerStatuses.sum { |cs| cs.restartCount.to_i }
 
+    ch[:last_state_terminated] = false
+    ch[:terminations] = []
+    pod.status.containerStatuses.each do |cs|
```

Comment:
The alternative way of doing this figuring out what termination events look like and monitor just those events.

See lines 178-195 above.